src/main/java/at/hannibal2/skyhanni/features/mining/ActualGemstonePowderDisplay.kt

```diff
@@ -0,0 +1,285 @@
+    }
+
+    private fun getEchoOfAtomizedBonusRate(): Double {
+        return 0.02 // 2%
+    }
+
+
+    @HandleEvent
+    fun onSecondPassed() {
+        if (!isEnabled()) return
+
+        is2xPowderActive = false
+
+        if (TabWidget.EVENT.isActive) {
+            for (line in TabWidget.EVENT.lines) {
+                if (line.contains("2x Powder")) {
+                    is2xPowderActive = true
+                    break
+                }
+            }
+        } else {
+            powderBossBarPattern.matchMatcher(BossbarData.getBossbar()) {
+                is2xPowderActive = true
+            }
+        }
+    }
+
+
+    private fun calculateMultiplierBreakdown(): MultiplierBreakdown {
+
+        var additiveBonus = 0.0
+
+        val atomizedCrystalsMulti = getAtomizedCrystalsBonus()
+        val echoOfAtomizedRate = getEchoOfAtomizedBonusRate()
+
+        additiveBonus += atomizedCrystalsMulti
+
+        val echoContribution = atomizedCrystalsMulti * echoOfAtomizedRate
+        additiveBonus += echoContribution
+
+        var totalMultiplier = 1.0 + additiveBonus
+
+        val is2xActive = is2xPowderActive
+        if (is2xActive) {
+            totalMultiplier *= 2.0
+        }
+
+        val isSkymallActive = HotmData.SKY_MALL.enabled && HotmApi.skymall == HotmApi.SkymallPerk.EXTRA_POWDER
+        if (isSkymallActive) {
+            totalMultiplier *= 1.15
+        }
+
+
+
+        val hotmBuffPercent = if (HotmData.POWDER_BUFF.enabled) {
+            HotmData.POWDER_BUFF.activeLevel
+        } else 0
+
+        val drillFractional = getDrillMultiplierFraction()
+        val drillAdditiveBonusPercent = ((drillFractional - 1) * 100.0)
+        ChatUtils.debug("[PowderDebug] drillAdditiveBonusPercent: $drillAdditiveBonusPercent")
+
+        val combinedAdditivePercent = hotmBuffPercent + drillAdditiveBonusPercent
+
+        ChatUtils.debug("[PowderDebug] combinedAdditivePercent: $combinedAdditivePercent")
+
+        if (combinedAdditivePercent > 0) {
+            totalMultiplier *= (1.0 + combinedAdditivePercent / 100.0)
+        }
+
+        val petFractional = getPetMultiplierFraction()
+        val petName = if (petFractional > 1.0) {
+            CurrentPetApi.currentPet?.cleanName
+        } else null
+
+        totalMultiplier *= petFractional
+
+        ChatUtils.debug("[PowderDebug] Total Multiplier: $totalMultiplier")
+
+        return MultiplierBreakdown(
+            atomizedCrystals = atomizedCrystalsMulti,
+            echoOfAtomizedRate = echoOfAtomizedRate,
+            is2xEventActive = is2xActive,
+            isSkymallActive = isSkymallActive,
+            hotmBuffPercent = hotmBuffPercent,
+            drillMultiplier = drillFractional,
+            petMultiplier = petFractional,
+            petName = petName,
+            totalMultiplier = totalMultiplier
+        )
+    }
+
+    private fun MultiplierBreakdown.toTooltipLines(baseAmount: Int, effectiveAmount: Int): List<String> {
```

Comment:
You can use a buildList here.
```suggestion
    private fun MultiplierBreakdown.toTooltipLines(baseAmount: Int, effectiveAmount: Int): List<String> = buildList {
```
Using that is the same as the lines. But ``lines.add(...)`` --> ``add(...)``. And you don't have to have a explicit return.